pkg/build/rpm_test.go

```diff
@@ -12,113 +13,252 @@ import (
 	require.NoError(t, err)
+	defer file2.Close()
+
+	RPMFileNames, err := getRPMFileNames(RPMSourceDir)
+	require.NoError(t, err)
 
 	// Test
-	rpmFileNames, err := builder.getRPMFileNames(rpmSourceDir)
+	err = copyRPMs(RPMSourceDir, builder.combustionDir, RPMFileNames)
 
 	// Verify
 	require.NoError(t, err)
 
-	assert.Contains(t, rpmFileNames, "rpm1.rpm")
-	assert.Contains(t, rpmFileNames, "rpm2.rpm")
+	_, err = os.Stat(filepath.Join(builder.combustionDir, "rpm1.rpm"))
+	require.NoError(t, err)
 
-	// Cleanup
-	assert.NoError(t, file1.Close())
-	assert.NoError(t, file2.Close())
+	_, err = os.Stat(filepath.Join(builder.combustionDir, "rpm2.rpm"))
+	require.NoError(t, err)
 }
 
-func TestCopyRPMs(t *testing.T) {
+func TestGetRPMFileNamesNoRPMs(t *testing.T) {
 	// Setup
-	bc := config.BuildConfig{
-		ImageConfigDir: "../config/testdata",
-	}
+	tmpDir, err := os.MkdirTemp("", "eib-copy-RPMs-test-no-RPMs")
+	require.NoError(t, err)
+	defer os.RemoveAll(tmpDir)
+
+	RPMSourceDir := filepath.Join(tmpDir, "rpms")
+	err = os.Mkdir(RPMSourceDir, 0o755)
+	require.NoError(t, err)
+
+	bc := config.BuildConfig{}
 	builder := New(nil, &bc)
-	err := builder.prepareBuildDir()
+	err = builder.prepareBuildDir()
 	require.NoError(t, err)
-	defer os.Remove(builder.eibBuildDir)
 
-	rpmSourceDir := filepath.Join(builder.buildConfig.ImageConfigDir, "rpms")
-	rpmDestDir := builder.combustionDir
+	// Test
+	RPMFileNames, err := getRPMFileNames(RPMSourceDir)
+
+	// Verify
+	require.ErrorContains(t, err, "no RPMs found")
+
+	assert.Empty(t, RPMFileNames)
+}
+
+func TestCopyRPMsNoRPMDir(t *testing.T) {
+	// Setup
+	tmpDir, err := os.MkdirTemp("", "eib-copy-RPMs-test-no-dir-")
+	require.NoError(t, err)
+	defer os.RemoveAll(tmpDir)
 
-	file1Path := filepath.Join(rpmSourceDir, "rpm1.rpm")
-	defer os.Remove(file1Path)
+	bc := config.BuildConfig{}
+	builder := New(nil, &bc)
+	err = builder.prepareBuildDir()
+	require.NoError(t, err)
+
+	RPMSourceDir, err := builder.generateRPMPath()
+	require.NoError(t, err)
+
+	// Test
+	err = copyRPMs(RPMSourceDir, builder.combustionDir, nil)
+
+	// Verify
+	require.NoError(t, err)
+}
+
+func TestWriteRPMScript(t *testing.T) {
+	// Setup
+	tmpDir, err := os.MkdirTemp("", "eib-write-RPM-script-test-")
+	require.NoError(t, err)
+	defer os.RemoveAll(tmpDir)
+
+	RPMSourceDir := filepath.Join(tmpDir, "rpms")
+	err = os.Mkdir(RPMSourceDir, 0o755)
+	require.NoError(t, err)
+
+	buildConfig := config.BuildConfig{}
+	builder := New(nil, &buildConfig)
+	require.NoError(t, builder.prepareBuildDir())
+
+	file1Path := filepath.Join(RPMSourceDir, "rpm1.rpm")
 	file1, err := os.Create(file1Path)
 	require.NoError(t, err)
+	defer file1.Close()
 
-	file2Path := filepath.Join(rpmSourceDir, "rpm2.rpm")
-	defer os.Remove(file2Path)
+	file2Path := filepath.Join(RPMSourceDir, "rpm2.rpm")
 	file2, err := os.Create(file2Path)
 	require.NoError(t, err)
+	defer file2.Close()
+
+	RPMFileNames, err := getRPMFileNames(RPMSourceDir)
+	require.NoError(t, err)
```

Comment:
Nit: The test aims to verify the logic of `writeRPMScript` and not of `getRPMFileNames` so we can just pass a pre-populated slice to `writeRPMScript`.